src/Pos/Explorer/Txp/Toil/Logic.hs

```diff
@@ -132,3 +150,59 @@ getTxRelatedAddrs (UnsafeTx {..}, _, _) undo =
     toSet = HS.fromList . toList
     -- Safe here, because union of non-empty sets can't be empty.
     unionNE lhs rhs = NE.fromList $ toList $ HS.union (toSet lhs) (toSet rhs)
+
+combineBalanceUpdates :: BalanceUpdate -> [(Address, (Sign, Coin))]
```

Comment:
I mean, why not change type of `minusBalance` from `[(Address, Coin)]` to `HashMap Address (Sign, Coin)`? Same for `plusBalance`. Probably it would simplify this code, but now I am looking at it and am not sure.